services/supervisor/src/errors.rs

```diff
@@ -15,39 +15,60 @@
 use exonum_derive::*;
 
 /// Common errors emitted by transactions during execution.
+///
+/// Errors are divided into sub-groups by the corresponding error codes ranges:
+/// - 0 - 15: Common `Supervisor` errors.
+/// - 16 - 31: Errors related to artifacts.
+/// - 32 - 47: Errors related to service instances.
+/// - 48 - 64: Errors related to configuration changes.
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 #[derive(ExecutionFail)]
 pub enum Error {
-    /// Artifact has been already deployed.
-    AlreadyDeployed = 0,
+    // # General errors that can occur within supervisor interaction.
```

Comment:
What do you think about splitting the error enum itself into several enums for different error categories?